gateway-provider-identity-assertion-common/src/main/java/org/apache/knox/gateway/identityasserter/common/filter/CommonIdentityAssertionFilter.java

```diff
@@ -84,57 +98,86 @@ public void init(FilterConfig filterConfig) throws ServletException {
         throw new ServletException("Unable to load principal mapping table.", e);
       }
     }
-    virtualGroupMapper = new VirtualGroupMapper(loadVirtualGroups(filterConfig));
-    String impersonationListFromConfig = filterConfig.getInitParameter(IMPERSONATION_PARAMS);
-    if (impersonationListFromConfig == null || impersonationListFromConfig.isEmpty()) {
-      impersonationListFromConfig = filterConfig.getServletContext().getInitParameter(IMPERSONATION_PARAMS);
-    }
-    initImpersonationParamsList(impersonationListFromConfig);
+
+    //subsequent iterations on filterConfig.getInitParameterNames() would not work if we simply used that as an enumeration
```

Comment:
I'll add a more detailed explanation in the code too: `getInitParameters()` returns an enumeration and the first time we iterate thru on its element we can process the parameter names (`hasMoreElements` returns `true`). The subsequent calls, however, will not succeed because `getInitParameters()` returns the same object where the `hasMoreElements` returns `false`. In this class, the first event is populating virtual group mapping and then we do the same for proxyuser config.